src/app/controllers/locations/edit.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+flnd.locationEdit = {
+    retrieveRecord: function(url, $scope, messenger, authService) {
+
+        authService.get(url)
+            .then(function success(response) {
+                $scope.location = response.data;
+
+            }, function error(response) {
+
+                 messenger.displayErrorResponse($scope, response);
+
+            });
+    },
+
+    configureSubmit: function(url, $scope, messenger, authService, $uibModalInstance){
+        return function() {
+            $scope.loading = true;
+
+            authService.put(url, $scope.location)
+                .then(function success() {
+
+                    $uibModalInstance.close();
+
+                }, function error(response) {
+
+                    messenger.displayErrorResponse($scope, response);
+                })
+                .finally(function() {
+                    $scope.loading = false;
+                });
+        };
+    }
+};
+
+/**
+ * @ngdoc function
+ * @name flightNodeApp.controller.location:LocationEditController
+ * @description
+ * # LocationEditController
+ * Controller for the edit work type page.
+ */
+angular.module('flightNodeApp')
+    .controller('LocationEditController',
+        ['$scope', '$http', '$log', '$location', 'messenger', 'authService', '$routeParams', 'config', 'id', '$uibModalInstance',
+            function ($scope, $http, $log, $location, messenger, authService, $routeParams, config, id, $uibModalInstance) {
+
+                if (!(authService.isAdministrator() ||
+                      authService.isCoordinator())) {
+                    $log.warn('not authorized to access this path');
+                    $location.path('/');
+                    return;
+                }
```

Comment:
I need to look into putting the above code into app.js's `.config()` call instead of embedding it inside each controller. Clearly a hacker could overcome this. But it will help keep casual users from trying to poke around in sections they shouldn't be able to access.